cmd/goupnpdcpgen/provider.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"archive/zip"
+	"fmt"
+	"path/filepath"
+)
+
+type dcpProvider interface {
+	process(tmpdir, name string, dcp *DCP) error
+}
+type upnpdotorg struct {
+	DocURL     string // Optional - URL for further documentation about the DCP.
+	XMLSpecURL string // Where to download the XML spec from.
+	// Any special-case functions to run against the DCP before writing it out.
+	Hacks []DCPHackFn
+}
+
+func (u upnpdotorg) process(tmpdir, name string, dcp *DCP) error {
+	dcp.DocURLs = append(dcp.DocURLs, u.DocURL)
+	specFilename := filepath.Join(tmpdir, name+".zip")
+	err := acquireFile(specFilename, u.XMLSpecURL)
+	if err != nil {
+		return fmt.Errorf("could not acquire spec for %s: %v", name, err)
```

Comment:
nit: prefer `%w` instead of `%v` for new calls to `fmt.Errorf` for the placeholder of `err`.

Reply:
(no need to do this for existing code, that would go way beyond the scope of this PR)